cmd/cmd.go

```diff
@@ -56,6 +56,12 @@ func ExecInCoreSnap() {
 
 	// should we re-exec? no option in the environment means yes
 	if !osutil.GetenvBool(key, true) {
+		logger.Debugf("re-exec disabled by user")
+		return
+	}
+
+	// did we already re-exec?
+	if osutil.GetenvBool("SNAP_DID_REEXEC") {
 		return
```

Comment:
Adding this would trigger a lot of additional output in our tests, each `snap foo` would print it which will break a lot of the tests that look for exact matches of output.